### PR TITLE
Allows copying an object and modifying metadata (user and object)

### PR DESCRIPTION
**Pull request by zack-shoylev**

### CODE REVIEW

**Comment by zack-shoylev** [comment]
https://issues.apache.org/jira/browse/JCLOUDS-872

File: apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/features/ObjectApi.java [hunk 1]
```diff
@@ -262,4 +263,36 @@ boolean copy(@PathParam("destinationObject") String destinationObject,
...
+   @Named("object:copy")
+   @PUT
+   @Path("/{destinationObject}")
+   @Headers(keys = OBJECT_COPY_FROM, values = "/{sourceContainer}/{sourceObject}")
+   @Fallback(FalseOnContainerNotFound.class)
+   boolean copy(@PathParam("destinationObject") String destinationObject,
+         @PathParam("sourceContainer") String sourceContainer,
+         @PathParam("sourceObject") String sourceObject,
+         @BinderParam(BindObjectMetadataToHeaders.class) Map<String, String> userMetadata,
+         @BinderParam(BindHeaderMetadataToHeaders.class) Map<String, String> objectMetadata);
```
Comment (andrewgaul) [inline apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/features/ObjectApi.java]: Should we reverse these parameters?  Also does this allow an easy way to copy the metadata from an existing object?  I would like to make the simple case of copying all objects from one container to another as easy as possible.

Comment (zack-shoylev) [inline apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/features/ObjectApi.java]: A very good point. I will investigate.

**Comment by zack-shoylev** [comment]
I have added a live test that demonstrates how the copy would work with real conditions.
Unfortunately, the content- metadata is handled separately and is also filtered from the headers. Which makes all of this very clunky! However, I did not want to address the header-filtering in this PR (or maybe at all).

**Comment by andrewgaul** [comment]
Pushed to master as a1cbec10925a05bf0a05c6ddb3f9eb51c7f155ea and 576005a33553d3b682bdd7088c10c5e7da1c004e.

**Pull request closed without merging**